### PR TITLE
Training discriminator loop/bill

### PR DESCRIPTION
**Pull request by billyang98**
added some more logic and support for the discriminator dataloader + training

### CODE REVIEW

**Comment by billyang98** [comment]
This branch is built on top of the Data splits training/bill branch, so look at that one first and then I'll merge this after that one is approved. This one is less prio.

File: src/train.py
Comment (Lucaskabela) [inline src/train.py:111]: Should we consider training the discriminator on a subset of the full_train_dataset?  Otherwise it will take quite a while.. maybe use `dataset = torch.utils.data.Subset(dataset, random_indices) ` to get a different random subset of say 1000 examples on each epoch

**Review by Lucaskabela — CHANGES_REQUESTED** [review]

**Review by Lucaskabela — APPROVED** [review]

File: src/train.py
Comment (liaojh1998) [reply src/train.py:111]: You don't have to create a separate Subset for this, you can simply use
```python3
discriminator_train_examples = 1000
discriminator_dataloader = DataLoader(full_train_dataset, batch_size=args.batch_size,
                                      sampler=np.random.choice(len(full_train_dataset), discriminator_train_examples),
                                      collate_fn=collate_fn_transformer, drop_last=True, num_workers=16)
```
This avoids having to have a new subset every epoch.

**Review by liaojh1998 — APPROVED** [review]

**Pull request merged**
